SPEC.md

```diff
@@ -0,0 +1,28 @@
+# Navigation.js specification
+
+## Purpose
+The goal of this specification is to outline the rules and ideas this library adheres to guide a user along a route.
+
+### Units
+By default, the metric system should be used where available.
+
+### Concepts
+Navigation.js looks at a single step along a greater route. All calculations are done against this step and not against the whole route. Once a step is completed, the next step on the route can be compared. If the user is no longer on the current step, the developer should re-route the user and also set their internal step counter back to 0.
+
+### Re-routing
+If at any time the user is not within `45 meters` of any point on the current step, re-route the user.
```

Comment:
I think this should vary by mode of transportation. If you're walking, you'll want to know when you walk half as many meters from the route; 45 meters can be significant ground to cover when you keep going off-route.